networkx/drawing/nx_pylab.py

```diff
@@ -470,6 +476,40 @@ def draw_networkx_nodes(
         labelleft=False,
     )
 
+    if auto_margins:
+        try:  # sets x and y margins separately if node_size is iterable
+            node_coords_sizes = list(zip(pos.items(), node_size))
+
+            x_min = sorted(node_coords_sizes, key=lambda a: a[0][1][0], reverse=False)[
+                0
+            ]
+            x_max = sorted(node_coords_sizes, key=lambda a: a[0][1][0], reverse=True)[0]
+            x_node_size = max([x_min, x_max], key=itemgetter(1))[1]
+
+            if x_node_size >= 500:
```

Comment:
Can you add comments explaining why the constants here are 100, 72, 50 and maybe 500 as well. If arbitrary or based on what seems to work well, then say that too.